L1Trigger/Phase2L1ParticleFlow/interface/Region.h

```diff
@@ -0,0 +1,104 @@
+#ifndef L1Trigger_Phase2L1ParticleFlow_Region_h
+#define L1Trigger_Phase2L1ParticleFlow_Region_h
+
+#include "L1Trigger/Phase2L1ParticleFlow/interface/DiscretePFInputs.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+namespace l1tpf_impl {
+  struct Region : public InputRegion {
+    std::vector<PFParticle> pf;
+    std::vector<PFParticle> puppi;
+    unsigned int caloOverflow, emcaloOverflow, trackOverflow, muonOverflow, pfOverflow, puppiOverflow;
+
+    const bool relativeCoordinates;  // whether the eta,phi in each region are global or relative to the region center
+    const unsigned int ncaloMax, nemcaloMax, ntrackMax, nmuonMax, npfMax, npuppiMax;
+    Region(float etamin,
+           float etamax,
+           float phicenter,
+           float phiwidth,
+           float etaextra,
+           float phiextra,
+           bool useRelativeCoordinates,
+           unsigned int ncalomax,
+           unsigned int nemcalomax,
+           unsigned int ntrackmax,
+           unsigned int nmuonmax,
+           unsigned int npfmax,
+           unsigned int npuppimax)
+        : InputRegion(0.5 * (etamin + etamax), etamin, etamax, phicenter, 0.5 * phiwidth, etaextra, phiextra),
+          relativeCoordinates(useRelativeCoordinates),
+          ncaloMax(ncalomax),
+          nemcaloMax(nemcalomax),
+          ntrackMax(ntrackmax),
+          nmuonMax(nmuonmax),
+          npfMax(npfmax),
+          npuppiMax(npuppimax) {}
+
+    enum InputType { calo_type = 0, emcalo_type = 1, track_type = 2, l1mu_type = 3, n_input_types = 4 };
+    static const char* inputTypeName(int inputType);
+
+    enum OutputType {
+      any_type = 0,
+      charged_type = 1,
+      neutral_type = 2,
+      electron_type = 3,
+      pfmuon_type = 4,
+      charged_hadron_type = 5,
+      neutral_hadron_type = 6,
+      photon_type = 7,
+      n_output_types = 8
+    };
+    static const char* outputTypeName(int outputType);
```

Comment:
> return std::string rather than c-string?

That would lead to additional allocations on the heap for no reason.
An alternative could be [`std::string_view`](https://en.cppreference.com/w/cpp/string/basic_string_view) if there is any need for a `std::string`-like interface.

Reply:
they're always compile-time constant strings, that's why I left them as const char *, but if you prefer std::strings returned by value I can change

Reply:
regarding the const, it's a static function, so I can't make it const

Reply:
> if you prefer std::strings returned by value

ehm, no - either `const char*` or `std::string_view`

Reply:
If I'm not mistaken, the `outputTypeName()` is used in string concatenation with `operator+()`, and unfortunately there is no overload of `operator+()` that would take `std::string` and `std::string_view`. Therefore `const char*` could be the most practical option.